### PR TITLE
chore: remove Observatorium DEX authentication support

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10990

Initially we used DEX based authentication when authenticating against Observatorium in KAS Fleet Manager and the Dataplanes.

Currently however we don't authenticate using DEX anymore but through OIDC.
Neither our Stage or Production environments uses DEX. The control plane development team does not make use of that authentication method either. The testing pipelines neither and we are not aware of anyone using it anymore.

Currently KFM code around Observatorium authentication is confusing and a maintenance burden due to the combination of OIDC and DEX. If we remove the DEX authentication support we will greatly simplify that area.

This PR takes care of that.

Specifically, the following KFM CLI flags are removed:
```
--dex-password-file
--dex-secret-file
--dex-url
--dex-username
--observatorium-gateway # This flag is only used for Observatorium DEX authentication
--observatorium-tenant # This flag is only used for Observatorium DEX authentication
--observatorium-token-file # This flag is only used for Observatorium DEX authentication
```

It also takes care of removing all the related code, OCP templates and Makefile contents that are associated/related to those flags

Additional remarks:
* This will need a post-cleanup of the KFM Stage and Production environments OCP template parameters
* This will need a post-cleanup of the KFM Stage and Production environments Vault secrets
* This needs some changes in the managed-kafka-versions repository. The changes in there should be rolled out before merging this PR. I am currently working on preparing those changes and I will notify it in this PR once that is ready
* This needs some changes in the KAS installer repository. The changes have already been made and rolled out there
* There is an additional CLI flag `--observatorium-auth-type` that with the removal of the `dex` authentication support only accepts the `redhat` value (for Red Hat SSO based auth). I intentionally left that flag only accepting one value for this PR. We can decide we either keep with the potential to be expanded later or we simply remove it to simplify further that area. I would like a discussion about it but let's do it through a separate channel and independently of this PR
* I have identified other flags that might be further removed but unrelated to DEX. I will open a different PR for that.

## Verification Steps

First of all, make sure in your local environment the following local secrets are removed:
```
secrets/dex.secret
secrets/dex.password
secrets/observatorium.token
```

* KAS Fleet Manager can run correctly
* Tests pass

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Review by machi1990 — APPROVED** [review]
reviewed the change, they look good to me

**Comment by machi1990** [comment]
> There is an additional CLI flag --observatorium-auth-type that with the removal of the dex authentication support only accepts the redhat value (for Red Hat SSO based auth). I intentionally left that flag only accepting one value for this PR. We can decide we either keep with the potential to be expanded later or we simply remove it to simplify further that area. I would like a discussion about it but let's do it through a separate channel and independently of this PR

the fact that the flag says values is `redhat` is confusing. People would expect something like `oidc` but that's a separate conversation and not related to what is being said here. What I really want to share is that, I think this flag could be removed. I am fine seeing that being evaluated and be done in another PR. 

> I have identified other flags that might be further removed but unrelated to DEX. I will open a different PR for that.

+1

**Comment by miguelsorianod** [comment]
> > There is an additional CLI flag --observatorium-auth-type that with the removal of the dex authentication support only accepts the redhat value (for Red Hat SSO based auth). I intentionally left that flag only accepting one value for this PR. We can decide we either keep with the potential to be expanded later or we simply remove it to simplify further that area. I would like a discussion about it but let's do it through a separate channel and independently of this PR
> 
> the fact that the flag says values is `redhat` is confusing. People would expect something like `oidc` but that's a separate conversation and not related to what is being said here. What I really want to share is that, I think this flag could be removed. I am fine seeing that being evaluated and be done in another PR.
> 
> > I have identified other flags that might be further removed but unrelated to DEX. I will open a different PR for that.
> 
> +1

Totally agree with the name `redhat` being confusing. Even the flag name I think it is. Let's discuss how we go ahead through other channels :+1:

File: pkg/client/observatorium/observability_config.go [hunk 1]
```diff
@@ -22,31 +21,19 @@ const (
...
-	AuthType             string        `json:"auth_type" yaml:"auth_type"`
-	AuthToken            string        `json:"auth_token"`
-	AuthTokenFile        string        `json:"auth_token_file"`
-	Cookie               string        `json:"cookie"`
-	Timeout              time.Duration `json:"timeout"`
-	Insecure             bool          `json:"insecure"`
-	Debug                bool          `json:"debug"`
-	EnableMock           bool          `json:"enable_mock"`
+	AuthType   string        `json:"auth_type" yaml:"auth_type"`
+	Cookie     string        `json:"cookie"`
```
Comment (pb82) [inline pkg/client/observatorium/observability_config.go:32]: Do we still need this? It looks like it was only an option when the auth type was Dex.

Comment (miguelsorianod) [reply pkg/client/observatorium/observability_config.go:32]: You are right :+1: . It is still referenced in the Observatorium client code but then it is not used anywhere. I will remove it.

Comment (miguelsorianod) [reply pkg/client/observatorium/observability_config.go]: Updated

**Comment by machi1990** [comment]
@miguelsorianod will you be able to have a look at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/actions/runs/4562814924/jobs/8050508553 ? Seems like CI is failing due to a linting issue on main

**Comment by machi1990** [comment]
discussing it via other channels

**Comment by machi1990** [comment]
Fixed in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1681

**Pull request merged**